x/concentrated-liquidity/fees.go

```diff
@@ -202,9 +224,16 @@ func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
 		return sdk.Coins{}, err
 	}
 
-	// We need to update the position's accumulator to the current fee growth outside
-	// before we claim rewards.
-	if err := feeAccumulator.SetPositionCustomAcc(positionKey, feeGrowthOutside); err != nil {
+	position, err := accum.GetPosition(feeAccumulator, positionKey)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	// prior to calling ClaimRewards (which updates unclaimed rewards), we must
+	// set initialFeeAccumulatorValue to initialFeeAccumulatorValue + feeGrowthOutside
+	customAccumulatorValue := position.InitAccumValue.Add(feeGrowthOutside...)
+	err = feeAccumulator.SetPositionCustomAcc(positionKey, customAccumulatorValue)
+	if err != nil {
 		return sdk.Coins{}, err
 	}
```

Comment:
Let's encapsulate this into a function with tests please

Reply:
https://github.com/osmosis-labs/osmosis/pull/4265/commits/59f795f6e0af364976af773c4c70025c6d2a1475